frontend/src/Components/OccupiedForm/OccupiedForm.tsx

```diff
@@ -0,0 +1,156 @@
+import { useEffect, useState } from 'react';
+import { OccupiedTimeSlotDto } from '~/dto';
+import styles from './OccupiedForm.module.scss';
+import { KEY } from '~/i18n/constants';
+import { toast } from 'react-toastify';
+import { getOccupiedTimeslots, postOccupiedTimeslots } from '~/api';
+import { useTranslation } from 'react-i18next';
+import { InputField } from '../InputField';
+import { Button } from '../Button';
+import { COLORS } from '~/types';
+import { IconButton } from '../IconButton';
+
+type OccupiedFormProps = {
+  recruitmentId: number;
+};
+
+type OccupiedLineProps = {
+  timeslot: OccupiedTimeSlotDto;
+  onChange: (key: number, newTimeslot: OccupiedTimeSlotDto) => void;
+  onDelete: (key: number) => void;
+  index: number;
+};
+
+function OccupiedLine({ timeslot, onChange, onDelete, index }: OccupiedLineProps) {
+  const { t } = useTranslation();
+
+  const [date, setDate] = useState<string>('');
+  const [startTime, setStartTime] = useState<string>('');
+  const [endTime, setEndTime] = useState<string>('');
+
+  const [dateError, setDateError] = useState<string>('');
+  const [startTimeError, setStartTimeError] = useState<string>('');
+  const [endTimeError, setEndTimeError] = useState<string>('');
+
+  useEffect(() => {
+    const startDt = new Date(timeslot.start_dt);
+    const endDt = new Date(timeslot.end_dt);
+    if (!isNaN(startDt.getTime())) {
+      setDate(startDt.toISOString().substring(0, 10));
+      setStartTime(startDt.toLocaleTimeString('no-NO').substring(0, 5));
+    }
+    if (!isNaN(startDt.getTime())) setEndTime(endDt.toLocaleTimeString('no-NO').substring(0, 5));
```

Comment:
Det er ikke så lett å se hva som skjer her. Prøv å gjøre det enklere?

Reply:
Sjekker om den allerede har start_dt, og konverterer den så til lokaltid for å unngå endringer som kan oppstå av det
Dermed får delen av stringen som kan brukes til date

Reply:
Det er vanskelig å vite hva dette er `.substring(0, 5)`
Gjerne refaktorer slik at det er enklere å lese.

Reply:
Kanskje legg til kommentarer på linjene:
39 //retrives substring "YYYY-MM-DD" from "YYYY-MM-DDTHH:mm:ss.sssZ"

40 //retrives substring "hh:mm" from "hh:mm:ss"

Reply:
Eller reformater koden ca. sånn:
```suggestion
  const formatDate = (date: Date) => date.toISOString().substring(0, 10);
  const formatTime = (date: Date) => date.toLocaleTimeString('no-NO').substring(0, 5);
  const startDt = new Date(timeslot.start_dt);
  const endDt = new Date(timeslot.end_dt);
  setDate(formatDate(startDt));
  setStartTime(formatTime(startDt));
  setEndTime(formatTime(endDt));
```

Reply:
Legger til kommentar, er ganske enkelt